grpc/client.go

```diff
@@ -77,20 +73,26 @@ type clientMetrics struct {
 }
 
 // NewClientMetrics constructs a *grpc_prometheus.ClientMetrics, registered with
-// the given registry, with timing histogram enabled. It must be called a
-// maximum of once per registry, or there will be conflicting names.
-func NewClientMetrics(stats registry) clientMetrics {
+// the given registry, with timing histogram enabled. If called more than once
+// on a single registry, it will gracefully avoid registering duplicate metrics.
+func NewClientMetrics(stats prometheus.Registerer) clientMetrics {
 	// Create the grpc prometheus client metrics instance and register it
 	grpcMetrics := grpc_prometheus.NewClientMetrics()
 	grpcMetrics.EnableClientHandlingTimeHistogram()
-	stats.MustRegister(grpcMetrics)
+	err := stats.Register(grpcMetrics)
+	if err != nil && !errors.As(err, &prometheus.AlreadyRegisteredError{}) {
```

Comment:
I would prefer if we continued to use `.MustRegister()` here. Perhaps I'm just misunderstanding something or wrong here but I believe the Register/MustRegister methods decide if a metric has or has not been "seen" based on an Id derived from the metric name. Just because we've already seen a metric name does not mean that our prometheus Registerer already holds a reference to that metric's underlying channel though. Ignoring this error could result in observations failing to be collected from those channels over time.

Reply:
Shoot, I think you're actually right. I wish that, instead of
```
foo = prometheus.Gauge(...)
prometheus.Register(foo)
```
it was more like
```
foo = prometheus.Instantiate(prometheus.GaugeDescription(...))
```
and then if you called Instantiate with the same description multiple times, it would just return a reference to the same underlying object every time. Problem gone!

I'll figure out what to do about this next week.

Reply:
Oh wild, it looks like the `prometheus.AlreadyRegisteredError` gives you a way to recover from this!
https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#example-AlreadyRegisteredError

How does this solution look to you?

Reply:
Omg... I hate how neat `are.ExistingCollector.(prometheus.Counter)` is.